app/models/ongoing_event.rb

```diff
@@ -0,0 +1,43 @@
+class OngoingEvent < ApplicationRecord
+  acts_as_paranoid
+
+  belongs_to :ngo
+  has_many   :participations, dependent: :destroy
+  has_many   :users,          through:   :participations
+
+  validates :title, length: { in: 1..100 }
+  validates :address, presence: true
+  validates :contact_person, presence: true
+
+  scope :newest_first, -> { order(:created_at) }
+  scope :published,    -> { where.not(published_at: nil) }
+
+  def pending?
```

Comment:
Besides that, if it makes sense to implement it in another way, totally go for it. I just put it like this, cause it was behaviorwise the closest I could get from refactoring some models using state machines a day before the release.